### ISSUE TITLE
How to upgrade to 5.x

### ISSUE THREAD

**Issue by tkruse** [wooorm/remark-lint#129]
Hi,
I have issues upgrading remark, remark-cli or remark-lint. Currently I use these:
```
    "dependencies": {
        "remark": "^4.2.2",
        "remark-lint": "^4.2.0",
        "remark-lint-sentence-newline": "^2.0.0",
        "remark-validate-links": "^4.1.0"
    }
```

And I get valid warnings e.g. for hello.md
```
## Hello world!

<b>bold</b>

***foo***
```

```
$ remark --use remark-lint hello.md 1> /dev/null 
hello.md
   1:1-1:16  warning  First heading level should be `1`     first-heading-level
   1:1-1:16  warning  Don’t add a trailing `!` to headings  no-heading-punctuation
    3:1-3:4  warning  Do not use HTML in markdown           no-html
   3:8-3:12  warning  Do not use HTML in markdown           no-html
       5:10  warning  Remove 1 line after node              no-consecutive-blank-lines
```

When I try to upgrade remark-lint to 5.x or 6.x, the same command outputs: "no issues found".
Same when I try using the latest remark-lint with remark-cli 3.0.0 (I already tried installing both recommended and consistent presets, and set them as presets (in remarkrc and using --use), and set them as plugins, none of which yielded warnings).

**Comment by wooorm** [comment]
Hi @tkruse!

Sorry about that, I didn’t have time to write that info yet. I just added the info to the [releases](https://github.com/wooorm/remark-lint/releases/tag/6.0.0) page.
I think that makes things more clear, but could you let me know if not, here or on Gitter?

**Comment by tkruse** [comment]
Hi Titus,

The releases page still does not help me much. My problem is that I do not get any warnings any more. E.g. see this example:

```
    $ npm install remark remark-lint remark-preset-lint-recommended remark-preset-lint-consistent
/scripts/nodejs
└── remark@7.0.0 

npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.1.1
$ remark --use lint --use remark-preset-lint-recommended --use remark-preset-lint-consistent hello.md 1> /dev/null 
hello.md: no issues found
```

There should be issues (see initial post), and remark-lint 4.x finds them. Do I need to pass each rule using --use? Or install more rules?

**Comment by wooorm** [comment]
Hi Tim! 👋

remark-lint 4x find those issues because it includes the rules you want (first-heading-level, no-heading-punctuation, etc.). remark-lint 6x does not find those issues, as it’s not configured to!

If you amend your `hello.md` like so:

```diff
 ## Hello world!

 <b>bold</b>

-***foo***
+***foo*** ___bar___
```

And now run remark on that (`remark . -u preset-lint-consistent`), you’ll get messages:

```txt
hello.md
  5:11-5:20  warning  Strong should use `*` as a marker    strong-marker    remark-lint
  5:13-5:18  warning  Emphasis should use `*` as a marker  emphasis-marker  remark-lint

⚠ 2 warnings
```

That works because remark-preset-lint-consistent includes those rules. If you want to enable different rules, you need to use them too (or create your own preset).

**Comment by tkruse** [comment]
I see. thanks, that helps a bit. I assumed all rules were included in the recommended set, that was my mistake. So now I  migrated my .remarkrc file. The documentation for that is also a bit weak. Before I had this:

```
// package.json   
    "dependencies": {
        "remark": "^4.2.2",
        "remark-lint": "^4.2.0",
        "remark-lint-sentence-newline": "^2.0.0",
        "remark-validate-links": "^4.1.0"
    }

// .remarkrc
{
    "presets": ["lint-recommended", "lint-consistent"],
    "plugins": {
        "remark-lint": {
            "unordered-list-marker-style": "consistent",
            "list-item-bullet-indent": true,
            "no-multiple-toplevel-headings": true,
            "maximum-line-length": 9000,
            "maximum-heading-length": 300,
            "no-tabs": true,
            ...
        }
    },
    "settings": {
        "bullet": "*",
        "listItemIndent": "1",
        "strong": "*",
        "emphasis": "*"
    }
```
Now with remark-lint 6.x, it looks like this
´´´
// package.json
"dependencies": {
    "remark-cli": "^3.0.0",
    "remark-lint": "^6.0.0",
    "remark-lint-first-heading-level": "^1.0.0",
    "remark-lint-list-item-bullet-indent": "^1.0.0",
    "remark-lint-maximum-line-length": "^1.0.0",
    "remark-lint-no-heading-punctuation": "^1.0.0",
    "remark-lint-unordered-list-marker-style": "^1.0.0",
    ...
    "remark-lint-sentence-newline": "^2.0.0",
    "remark-preset-lint-consistent": "2.0.0",
    "remark-preset-lint-recommended": "2.0.0",
    "remark-validate-links": "^6.0.0"
  }

// .remarkrc
{
    "plugins": {
        "remark-preset-lint-recommended": true,
        "remark-preset-lint-consistent": true,
        "remark-lint-no-heading-punctuation": false,
        "remark-lint-maximum-line-length": 40,
        ...
    },
    "settings": {
        "bullet": "*",
        "listItemIndent": "1",
        "strong": "*",
        "emphasis": "*"
    }
´´´
That was quite painful to find out for me, maybe I was not seeing the obvious. I recommend you write a small migration guide, with the necessary steps, e.g.:

* in remarkrc, convert presets to plugins
* in remarkrc, extract remark-lint rule configuration as plugins (if desired)
* in package.json, install all individual plugins and presets (used in remarkrc), name is prefixed with "remark-lint-" (unless installed by presets?)
* give small example of new structure 

It is quite painful to migrate, because normally a project will already have markdown files satisfying most rules, so to confirm that after the migration, the configuration still works, one would have to manually try violating each rule and check that this still generates a warning. (That way I just found out in my 4.x version, ```"no-multiple-toplevel-headings": true, had no effect```).

Another thing I notice is that remark will not produce an error for adding a disabled non-existing plugin, like:
```
       // in .remarkrc
        "remark-lint-hsadgafdfdsa": false,
```
that is a little annoying when trying to make sure preset rules are being disabled.

Is there a way to run remark and print a normalized configuration? Such as what individual rules are provided by presets, which rules are activated after applying presets and other rules?

Also I tried to install 

* vhf/remark-lint-no-empty-sections 
* vhf/remark-lint-no-url-trailing-slash 

But those did not generate any warnings for me(did not try with 4.x). Do they need to be changed to work with remark > 5.x? (might be nice to notify the authors of plugins).

Also I see on npm that there are several more rule packages by you which seem not to be included in the presets, nor mentioned in the remark-lint README, in particular regarding definitions and tables.
What I would love to do is to enable a large such set of rules for a project, then see by the warnings which rules would be useful to include. Is there any easy way to get that (e.g. a remark-lint-preset-all-wooorm)?

**Comment by wooorm** [comment]
> So now I migrated my .remarkrc file. The documentation for that is also a bit weak.

I’ve updated the releases page with some examples. I’d gladly accept a PR on docs or an issue for more if you have suggestions.

> Is there a way to run remark and print a normalized configuration? Such as what individual rules are provided by presets, which rules are activated after applying presets and other rules?

Use the [`DEBUG="*"`](https://github.com/unifiedjs/unified-args#debugging) env variable.

> might be nice to notify the authors of plugins

That’s exactly what I’ve been doing the last two weeks.
I’ve also [mentioned](https://github.com/wooorm/remark-lint#list-of-external-rules) which rules are not updated yet.

> Also I see on npm that there are several more rule packages by you which seem not to be included in the presets, nor mentioned in the remark-lint README, in particular regarding definitions and tables.

Rules are listed [here](https://github.com/wooorm/remark-lint#rules).

> What I would love to do is to enable a large such set of rules for a project, then see by the warnings which rules would be useful to include. Is there any easy way to get that (e.g. a remark-lint-preset-all-wooorm)?

How would that deal with configurable rules?
I think that’s a good idea. Would you be interested in creating a preset for that?

**Comment by tkruse** [comment]
thanks for the doc updates. I wish I had the tie to help coding, but sadly, I don't.

**Comment by wooorm** [comment]
Cheers! Contributions to docs are also greatly appreciated!
